File: Day<03>/Solution.py
class Solution:
    def smallestpositive(self, array, n):
        array.sort()
        result = 1
        for i in range(n):
            if(array[i] <= result):
                result = result + array[i]
        return result

if __name__ == '__main__':
    t = int(input())
    for _ in range(t):
        n = int(input())
        array = list(map(int, input().strip().split()))
        ob = Solution()
        print(ob.smallestpositive(array,n))

File: Day<03>/Solution.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution
{
    public:
    long long smallestpositive(vector<long long> array, int n)
    {
        sort(array.begin(),array.end());
        long long result = 1;
        for (int i = 0; i < n && array[i] <= result; i++)
            result = result + array[i];
        return result;
    }
};

int main()
{
    int t;
    cin >> t;
    while (t--)
    {
        int n;
        cin >> n;
        vector<long long> array(n);
        for (int i=0; i<n; ++i)
        {
            cin >> array[i];
        }
        Solution ob;
        cout << ob.smallestpositive(array,n) << "\n";
    }
    return 0;
}
